nomad/server.go

```diff
@@ -344,6 +346,7 @@ func NewServer(config *Config, consulCatalog consul.CatalogAPI, consulConfigEntr
 		blockedEvals:     NewBlockedEvals(evalBroker, logger),
 		rpcTLS:           incomingTLS,
 		aclCache:         aclCache,
+		workersEventCh:   make(chan interface{}),
```

Comment:
Since you don't block on sending I think we should buffer at least 1 event to ensure at least 1 event is captured even when the receiver isn't ready.

```suggestion
		workersEventCh:   make(chan interface{}, 1),
```